nic_operator/nic_operator_ci_test.sh

```diff
@@ -283,6 +283,65 @@ function test_ofed_and_rdma {
     fi
 }
 
+function test_nv_peer_mem_with_host_device {
+    local resource_name_prefix="nvidia.com"
+    local resource_name="hostdev"
+
+    status=0
+
+    deploy_gpu_operator
+    let status=$status+$?
+    if [ "$status" != 0 ]; then
+        echo "Error: error in deploying the GPU operator."
+        return $status
+    fi
+
+    local sample_file="$ARTIFACTS"/ofed-host-device-nv-peer-nic-cluster-policy.yaml
+
+    configure_common "$sample_file"
+    configure_ofed "$sample_file"
+    configure_host_device "$sample_file" "$resource_name_prefix" "$resource_name"
+    configure_nv_peer_mem "$sample_file"
+
+    nic_policy_create "$sample_file"
+    let status=status+$?
+    if [ "$status" != 0 ]; then
+        echo "Error: error in creating the example nic_policy."
+        return $status
+    fi
+
+    sleep 10
+
+    test_ofed_drivers
+    let status=status+$?
+    if [ "$status" != 0 ]; then
+        echo "Error: Ofed modules failed!"
+        return $status
+    fi
+
+    test_gpu_direct "harbor.mellanox.com/cloud-orchestration/cuda-perftest:Ubuntu20.04-cuda-devel-11.2.1"\
```

Comment:
I'm wondering if we want to publish this image to some public registry

Reply:
we got this project with example Dockerfiles: https://github.com/Mellanox/k8s-images/tree/master/perftest
however as this is used primarily for testing there was no request to have a CI/CD pipe to publish pre-build images.

the cuda base image is available in dockerhub where users can use to build their cuda containers.

Reply:
Fair enough. LGTM